test/runtests.jl

```diff
@@ -125,88 +123,84 @@ function test_padm2(n::Int)
 
     A = sprand(n,n,0.0005) + 1im*sprand(n,n,0.0005)
 
-    tic()
-    w1 = padm(A)
-    t1 = toc()
+    t1 = @elapsed w1 = padm(A)
 
-    tic()
-    w2 = expm(full(A))
-    t2 = toc()
+    t2 = @elapsed w2 = expm_higham(Matrix(A))
 
     return norm(w1-w2)/norm(w2), t1, t2
 end
 
 println("testing real n=1000 (first padm, then expm)")
 res, t1, t2 = test_padm(1000)
-println("residuum: $res\n")
+println("residuum: $res")
+println("time dif: $(t1-t2)\n")
 @test res < 1e-6
 
 println("testing complex n=100 (first padm, then expm)")
 res, t1, t2 = test_padm2(1000)
-println("residuum: $res\n")
+println("residuum: $res")
+println("time dif: $(t1-t2)\n")
 @test res < 1e-6
 
 function test_chbv(n::Int)
 
     p = 0.1
-    D = diagm(-rand(n))
+    D = Matrix(Diagonal(-rand(n)))
     T = sprandn(n, n, p)
-    H = T * D * T.'  # random negative semidefinite symmetric matrix
+    H = T * D * transpose(T)  # random negative semidefinite symmetric matrix
+    
     vec = randn(n)
-    w1 = chbv(H, vec);
-
-    tic()
     w1 = chbv(H, vec)
-    t1 = toc()
+    t1 = @elapsed w1 = chbv(H, vec)
 
-    w2 = expm(full(H)) * vec
-    tic()
-    w2 = expm(full(H)) * vec
-    t2 = toc()
+    w2 = expm_higham(Matrix(H)) * vec
+    t2 = @elapsed  w2 = expm_higham(Matrix(H)) * vec
 
     return norm(w1-w2)/norm(w2), t1, t2
 end
 
 function test_chbv2(n::Int)
 
     p = 0.1
-    D = diagm(-rand(n))
+    D = Matrix(Diagonal(-rand(n)))
     T = sprandn(n, n, p) + sprandn(n, n, p)*im
-    H = T * D * T'  # random negative semidefinite hermitian matrix
+    H = T * D * adjoint(T)  # random negative semidefinite hermitian matrix
+    
     vec = randn(n) + randn(n)*im
-    w1 = chbv(H, vec);
-
-    tic()
     w1 = chbv(H, vec)
-    t1 = toc()
+    t1 = @elapsed w1 = chbv(H, vec)
 
-    w2 = expm(full(H)) * vec
-    tic()
-    w2 = expm(full(H)) * vec
-    t2 = toc()
+    w2 = expm_higham(Matrix(H)) * vec
+    t2 = @elapsed w2 = expm_higham(Matrix(H)) * vec
 
     return norm(w1-w2)/norm(w2), t1, t2
 end
 
 println("testing real n=100 (first chbv, then expm)")
 res, t1, t2 = test_chbv(100)
-println("residuum: $res\n")
+println("residuum: $res")
+println("time dif: $(t1-t2)\n")
 @test res < 1e-6
 
 println("testing complex n=100 (first chbv, then expm)")
 res, t1, t2 = test_chbv2(100)
-println("residuum: $res\n")
+println("residuum: $res")
+println("time dif: $(t1-t2)\n")
 @test res < 1e-6
 
 println("testing real n=1000 (first chbv, then expm)")
 res, t1, t2 = test_chbv(1000)
-println("residuum: $res\n")
+println("residuum: $res")
+println("time dif: $(t1-t2)\n")
 @test res < 1e-6
 
+#= commented since this test takes sometime difs very long
```

Comment:
Sorry! Yes, this took much longer, but I didn't investigate what causes it. ("difs" is just a weird typo) Since the other tests for `chbv` are Ok, I took this one out.